crates/re_chunk/src/transport.rs

```diff
@@ -0,0 +1,692 @@
+/// This has a stable ABI! The entire point of this type is to allow users to send raw arrow data
+/// into Rerun.
+/// This means we have to be very careful when checking the validity of the data: slipping corrupt
+/// data into the store could silently break all the index search logic (e.g. think of a chunk
+/// claiming to be sorted while it is in fact not).
+//
+// TODO(#4184): Provide APIs in all SDKs to log these all at once (temporal batches).
+#[derive(Debug)]
+pub struct TransportChunk {
+    /// The schema of the dataframe, and all chunk-level and field-level metadata.
+    ///
+    /// Take a look at the `TransportChunk::CHUNK_METADATA_*` and `TransportChunk::FIELD_METADATA_*`
+    /// constants for more information about available metadata.
+    pub schema: ArrowSchema,
+
+    /// All the control, time and component data.
+    pub data: ArrowChunk<Box<dyn ArrowArray>>,
+}
+
+impl std::fmt::Display for TransportChunk {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        re_format_arrow::format_dataframe(
+            &self.schema.metadata,
+            &self.schema.fields,
+            self.data.iter().map(|list_array| &**list_array),
+        )
+        .fmt(f)
+    }
+}
+
+impl TransportChunk {
+    /// The key used to identify a Rerun [`ChunkId`] in chunk-level [`ArrowSchema`] metadata.
+    pub const CHUNK_METADATA_KEY_ID: &'static str = "rerun.id";
+
+    /// The key used to identify a Rerun [`EntityPath`] in chunk-level [`ArrowSchema`] metadata.
+    pub const CHUNK_METADATA_KEY_ENTITY_PATH: &'static str = "rerun.entity_path";
+
+    /// The marker used to identify whether a chunk is sorted in chunk-level [`ArrowSchema`] metadata.
+    ///
+    /// The associated value is irrelevant -- if this marker is present, then it is true.
+    pub const CHUNK_METADATA_MARKER_IS_SORTED: &'static str = "rerun.is_sorted";
+
+    /// The key used to identify the kind of a Rerun column in field-level [`ArrowSchema`] metadata.
+    ///
+    /// That is: control columns (e.g. `row_id`), time columns or component columns.
+    pub const FIELD_METADATA_KEY_KIND: &'static str = "rerun.kind";
+
+    /// The value used to identify a Rerun time column in field-level [`ArrowSchema`] metadata.
+    pub const FIELD_METADATA_VALUE_KIND_TIME: &'static str = "time";
+
+    /// The value used to identify a Rerun control column in field-level [`ArrowSchema`] metadata.
+    pub const FIELD_METADATA_VALUE_KIND_CONTROL: &'static str = "control";
+
+    /// The value used to identify a Rerun data column in field-level [`ArrowSchema`] metadata.
+    pub const FIELD_METADATA_VALUE_KIND_DATA: &'static str = "data";
+
+    /// The marker used to identify whether a column is sorted in field-level [`ArrowSchema`] metadata.
+    ///
+    /// The associated value is irrelevant -- if this marker is present, then it is true.
+    pub const FIELD_METADATA_MARKER_IS_SORTED: &'static str = Self::CHUNK_METADATA_MARKER_IS_SORTED;
```

Comment:
I don't understand how this would be used in practice. All columns must have the same sort order within a chunk. This value seems like it would always have to match `CHUNK_METADATA_MARKER_IS_SORTED`

Reply:
Ahh, I think I understand, but could probably use an improved name.

It seems like:
 - CHUNK_METADATA_MARKER_IS_SORTED => This chunk is sorted by row-id
 - FIELD_METADATA_MARKER_IS_SORTED => This chunk is sorted on this timeline
 
 If all timelines are monotically increasing they all might be set, but it's possible to be sorted by row-id, but not sorted by timeline when out-of-order logging shows up.

Reply:
Yes, that's exactly what it is. I'll see if i can improve the naming/docs.